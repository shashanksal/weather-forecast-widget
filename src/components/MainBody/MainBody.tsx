import React from 'react';

//Material UI Components
import {
    Paper,
    Grid
} from '@material-ui/core';

//Custom component import
import LocationForm from '../LocationForm';
import WeatherWidget from '../WeatherWidget';
import Image from '../../assets/pexels-kaique-rocha-125510.jpg';

const styles = {
    paperContainer: {
        backgroundImage: `url(${Image})`,
        height: '100vh',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover'
    },
    content: {
        height: '100%',
        width: '100%',
        backgroundColor: 'rgba(0, 0, 0, 0.7)',
    }
}

const MainBody = (props: any) => {

    return (
        <Paper style={styles.paperContainer}>
            <div style={styles.content}>
                <Grid
                    container
                    spacing={0}
                    direction="column"
                    alignItems="center"
                    justify="center"
                    style={{ minHeight: '50vh' }}
                    className="containier-gridbox"
                >
                    <Grid item xs={5}>
                        <LocationForm></LocationForm>

                    </Grid>
                </Grid>
                <Grid
                    container
                    direction="column"
                    alignItems="center"
                    justify="center"
                >
                    <Grid item xs={8}>
                        <WeatherWidget></WeatherWidget>
                    </Grid>

                </Grid>
            </div>
        </Paper>
    )
}

export default MainBody;