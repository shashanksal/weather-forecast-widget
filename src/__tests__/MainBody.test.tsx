import React from "react";
import { render } from "@testing-library/react";

import MainBody from "../components/MainBody";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk, { ThunkDispatch } from "redux-thunk";

type DispatchExts = ThunkDispatch<any, void, any>;
const middlewares = [thunk];
const mockStore = configureMockStore<any, DispatchExts>(middlewares);
const mStore = mockStore();

describe("test category", () => {
	it("MainBody matches snapshot", () => {
		const { asFragment } = render(
			<Provider store={mStore}>
				<MainBody />
			</Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
