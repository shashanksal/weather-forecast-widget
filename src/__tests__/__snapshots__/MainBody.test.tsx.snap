// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test category MainBody matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style="background-image: url(); height: 100vh; background-position: center; background-repeat: no-repeat; background-size: cover;"
  >
    <div
      style="height: 100%; width: 100%; background-color: rgba(0, 0, 0, 0.7);"
    >
      <div
        class="MuiGrid-root containier-gridbox MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        style="min-height: 50vh;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <div
            style="width: 600px; height: 150px; display: flex; align-items: center; justify-content: center; background-color: gold; border: 2px solid #fff; border-radius: 4px;"
          >
            <form
              autocomplete="off"
              class="weather-form"
              novalidate=""
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                style="width: 400px; margin: 8px;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                  data-shrink="true"
                  for="standard-full-width"
                  id="standard-full-width-label"
                >
                  Location
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="standard-full-width"
                    placeholder="Enter Location"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained submit-btn MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  SEARCH
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <div
            style="width: 800px; height: auto; display: flex; align-items: center; justify-content: center;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
